python/lsst/daf/butler/registry/_config.py

```diff
@@ -85,7 +86,17 @@ def replaceRoot(self, root: str):
         """Replace any occurrences of `BUTLER_ROOT_TAG` in the connection
         with the given root directory.
         """
-        self["db"] = replaceRoot(self["db"], root)
+        # Do not examine URI if there is nothing to replace
+        if BUTLER_ROOT_TAG not in self["db"]:
+            return
+
+        # This can currently only work with file scheme
+        uri = ButlerURI(root)
+        if not uri.scheme or uri.scheme == "file":
+            path = uri.path
+        else:
+            raise RuntimeError(f"Root replacement only works for file URIs")
+        self["db"] = replaceRoot(self["db"], path)
```

Comment:
This was needed because of the `.geturl()` change. When the output config file was a relative path it was all fine but in the tests it's an absolute path and that means that `root` is set to `file://` and `replaceRoot` does not understand URIs (so it ended up with `sqlite:file:...` which does not work).  I made it part of this commit because I want tests to pass on this commit and not defer fixing the tests until a later commit. I realize that we aren't entirely clear on whether we prefer tests to pass over "separation of concerns" in commits.

Reply:
I think the former is "obviously" correct. If one commit breaks the build and another fixes it, then the commits are no longer self-contained changes.

Reply:
I did think about moving it into replaceRoot but for a datastore you generally will want file: and s3: to work as the root.